src/labone/nodetree/node.py

```diff
@@ -219,29 +224,25 @@ def __init__(
             paths_as_segments,
         )  # type: ignore[assignment]
 
-    def construct_nodetree(
-        self,
-        *,
-        hide_kernel_prefix: bool = True,
-    ) -> Node:
+        has_common_prefix = len(self._partially_explored_structure.keys()) == 1
+
+        if not hide_kernel_prefix or not has_common_prefix:
+            self._root_prefix = ()
+        else:
+            common_prefix = next(iter(self._partially_explored_structure.keys()))
+            self._root_prefix = (common_prefix,)
+
+    def construct_nodetree(self) -> Node:
```

Comment:
suggestion: I would just make this a property called `root` as well. Makes it easier to understand since it does not really construct the node tree but just accesses the first element.